ktor-server/ktor-server-core/jvm/src/io/ktor/routing/RouteSelector.kt

```diff
@@ -259,6 +329,28 @@ data class PathSegmentOptionalParameterRouteSelector(val name: String, val prefi
     }
 
     override fun toString(): String = "${prefix ?: ""}{$name?}${suffix ?: ""}"
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
I guess all equals and hashcodes should have tests

Reply:
same here